cmd/queue/main.go

```diff
@@ -219,6 +249,14 @@ func knativeProbeHandler(healthState *health.State, prober func() bool, isAggres
 			return
 		}
 
+		if preferPodForScaledown(logger, env.DownwardAPILabelsPath) {
+			//Deliberately failing the readiness probe when pod is labelled for scale down
+			http.Error(w, failingHealthcheck, http.StatusBadRequest)
```

Comment:
Why 400? Rather than say 500?

Reply:
400 has a more deliberate connotation to it.

Reply:
Dunno, 400 is client side error :-) But probably doesn't matter in this particular case.